src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -215,6 +222,14 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 		return providerId;
 	}
 
+	/**
+	 * Sets the provider ID for a URI
+	 */
+	public setProviderIdForUri(ownerUri: string, providerName: string): void {
```

Comment:
I'm not really understanding the need for both this method and `doChangeLanguageFlavor` (or at least why this is public)?  They seem to be serving the same purpose, though with different implementations, if I'm understanding the use-case correctly.  i.e. can't you just update this map from that method, or is there some behavior\use-case I'm missing?

Reply:
doChangeLanguageFlavor just fires this event and does not update the map: 

this._onLanguageFlavorChanged.fire({
				uri: uri,
				language: language,
				flavor: provider
			});

This map is used to maintain the providerId for URIs (either by connection or if user selected it manually). When we connect to a server in sendConnectRequest, we explicitly set: this._uriToProvider[uri] = connection.providerName;
So, we need to do something similar when user just changes the flavor so I added this method. 
Any other way we should do this?

Reply:
I was thinking that doChangeLanguageFlavor could update the map.  Is there any case where we would want to fire the event and not update the map, or update the map and not fire the event?  It seems like these should be coupled?

Reply:
I think before we were only updating the map when there was a connection and for other offline files we would just set this:  this.doChangeLanguageFlavor(uri, 'sql', defaultProvider); and not really put it in the map. So I was doing something similar where we only add it to the map when either we connect the script or explicitly change the flavor. For other default files we would just call ensureDefaultLanguageFlavor which would just call doChangeLanguageFlavor and not set the map.
I guess maybe now that we are adding offline files to map, we can do it for all of them either explicit or default. Does that sound okay?

Reply:
I would expect doChangeLanguageFlavor to update the provider ID map for that file, otherwise, it seems like things can get out of sync.

Reply:
Makes sense. I will make the change and test it to make sure I am not missing something.